### PR TITLE
Checkbox loading and disabled props implemented

### PR DESCRIPTION
**Pull request by dancingshell**
### Summary

* added checkbox spinner
* added disabled styles
(checkbox is disabled when disabled or loading are true)

> loading 
screen shot 2017-10-18 at 5 11 05 pm
> disabled
screen shot 2017-10-18 at 5 11 33 pm
> default
screen shot 2017-10-18 at 5 11 48 pm

### CODE REVIEW

File: checkbox/src/Checkbox.scss [hunk 1]
```diff
@@ -1,6 +1,11 @@
 @import '../../shared/variables';
 
 .container {
+  width: auto;
```
Comment (ben-burlingham) [inline checkbox/src/Checkbox.scss]: Width prop not needed?

File: checkbox/src/Checkbox.scss [hunk 2]
```diff
@@ -17,3 +22,37 @@
     visibility: visible;
   }
 }
+
+.spinnerContainer {
+  display: flex;
+  padding-left: 8px;
+  align-items: center;
+  justify-content: center;
```
Comment (ben-burlingham) [inline checkbox/src/Checkbox.scss]: Just FYI there's an SCSS lint file available too...alphabetization is pretty helpful for future you

File: checkbox/src/Checkbox.scss (hunk 1 above)
Comment (dancingshell) [reply checkbox/src/Checkbox.scss]: True now that I added the `display: flex;` I don't need the other default

File: checkbox/src/Checkbox.scss (hunk 2 above)
Comment (dancingshell) [reply checkbox/src/Checkbox.scss]: mmm interesting, i usually order them by html tree structure, but I cam see alphabetized being useful too

File: checkbox/test/Checkbox.test.jsx [hunk 3]
```diff
@@ -62,6 +62,15 @@ describe('Checkbox', () => {
       });
     });
 
+    describe('When loading is true', () => {
+      const disabledCheckboxWrapper = shallowRenderCheckbox({ loading: true });
```
Comment (mcclayton) [inline checkbox/test/Checkbox.test.jsx]: `nb`: Perhaps rename this variable. `disabledCheckboxWrapper` -> `loadingCheckboxWrapper`

File: checkbox/test/Checkbox.test.jsx [hunk 4]
```diff
@@ -88,4 +97,18 @@ describe('Checkbox', () => {
...
+
+  describe('Spinner', () => {
+    const loadingCheckbox = shallowRenderCheckbox({ loading: true });
+    const spinnerWrapper = loadingCheckbox.find('.spinnerContainer');
+
+    it('should be a span', () => {
+      expect(spinnerWrapper.type()).toBe('span');
+    });
+
+    it('should contain an span', () => {
```
Comment (mcclayton) [inline checkbox/test/Checkbox.test.jsx]: `nb`: `an` -> `a`

**Pull request merged**